tests/chainerx_tests/unit_tests/routines_tests/test_math.py

```diff
@@ -2118,6 +2209,42 @@ def func(self, xp, a):
         return xp.arctan(a)
 
 
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize(*(
+    # Special shapes
+    chainer.testing.product({
+        'in_shapes': _shapes_combination_binary,
+        'in_dtypes,out_dtype': (
+            _make_same_in_out_dtypes(2, chainerx.testing.float_dtypes)),
+        'input_lhs': [-3, -0.75, -0.4, 0.4, 0.75, 3],
+        'input_rhs': [-3, -0.75, -0.4, 0.4, 0.75, 3],
```

Comment:
Could you split the test for input values into another product (to reduce the number of test cases)?

Reply:
How does something like this look
```python
def differentiable_points_atan2(input_lhs, input_rhs):
    return chainer.testing.product({
        'in_shapes': [((2, 3), (2, 3))],
        'in_dtypes,out_dtype': (
            _make_same_in_out_dtypes(2, chainerx.testing.float_dtypes)),
        'input_lhs': input_lhs,
        'input_rhs': input_rhs,
    })


# Since the gradient of arctan2 is quite flaky.
# for smaller values especially `float16`.
@op_utils.op_test(['native:0', 'cuda:0'])
@chainer.testing.parameterize(*(
    # Special shapes
    chainer.testing.product({
        'in_shapes': _shapes_combination_binary,
        'in_dtypes,out_dtype': (
            _make_same_in_out_dtypes(2, chainerx.testing.float_dtypes)),
        'input_lhs': ['random'],
        'input_rhs': ['random'],
        'skip_backward_test': [True],
        'skip_double_backward_test': [True],
    })
    # Differentiable points
    + differentiable_points_atan2([-3.], [-3.])
    + differentiable_points_atan2([-1.], [-1.])
    + differentiable_points_atan2([1.], [1.])
    + differentiable_points_atan2([3.], [3.])
.....
```

Reply:
Sorry, I meant that I want to split the test into mixed-shape-test and input-value-combination-test.

Reply:
I am a bit confused about that, could you please give me an example.

Reply:
I expected the code like the following:
```py
@op_utils.op_test(['native:0', 'cuda:0'])
@chainer.testing.parameterize(*(
    # Mixed shapes
    chainer.testing.product({
        'in_shapes': _shapes_combination_binary,
        'in_dtypes,out_dtype': (
            _make_same_in_out_dtypes(2, chainerx.testing.float_dtypes)),
        'input_lhs': [1],
        'input_rhs': [1],
    })
    # Differentiable points
    chainer.testing.product({
        'in_shapes': [((2, 3), (2, 3))],
        'in_dtypes,out_dtype': (
            _make_same_in_out_dtypes(2, chainerx.testing.float_dtypes)),
        'input_lhs': [-3, -1, 1, 3],
        'input_rhs': [-3, -1, 1, 3],
    })

.....
```

Reply:
Oh. Thanks. Will copy this itself.